src/DynamoCore/Visualization/DefaultGraphicPrimitives.cs

```diff
@@ -0,0 +1,99 @@
+                ColorsStride = package.ColorsStride.ToString();
+            }
+
+            PointVertices = EncodeNumbers(package.PointVertices);
+            PointVertexColors = EncodeNumbers(package.PointVertexColors);
+
+            TriangleVertices = EncodeNumbers(package.MeshVertices);
+            TriangleNormals = EncodeNumbers(package.MeshNormals);
+            TriangleVertexColors = EncodeNumbers(package.MeshVertexColors);
+
+            LineStripVertices = EncodeNumbers(package.LineStripVertices);
+            LineStripCounts = EncodeNumbers(package.LineStripVertexCounts);
+            LineStripColors = EncodeNumbers(package.LineStripVertexColors);
+
+            RequiresPerVertexColoration = package.RequiresPerVertexColoration;
+        }
+
+        // Base-64 encoded array of 32 bit floats, 3 per vertex.
+        public string TriangleVertices { get; private set; }
+
+        // Base-64 encoded array of 32 bit floats, 3 per vertex.
+        public string TriangleNormals { get; private set; }
+
+        // Base-64 encoded array of 32 bit unsigned integers, 1 per vertex, in RGBA format.
+        public string TriangleVertexColors { get; private set; }
+
+        // Base-64 encoded array of 32 bit floats, 2 per vertex.
+        public string TriangleTextureCoordinates { get; private set; }
+
+        // Base-64 encoded array of 32 bit floats, 3 per vertex.
+        public string LineStripVertices { get; private set; }
+
+        // Base-64 encoded array of 32 bit unsigned integers, 1 per line strip, giving the number of vertices in the strip.
+        public string LineStripCounts { get; private set; }
+
+        // Base-64 encoded array of 32 bit unsigned integers, 1 per vertex, in RGBA format.
+        public string LineStripColors { get; private set; }
+
+        // Base-64 encoded array of 32 bit floats, 3 per vertex.
+        public string PointVertices { get; private set; }
+
+        // Base-64 encoded array of 32 bit unsigned integers, 1 per vertex, in RGBA format.
+        public string PointVertexColors { get; private set; }
+
+        // Base-64 encoded array of 32 bit unsigned integers in RGBA format, definining a texture to apply to the triangles.
+        public string Colors { get; private set; }
+
+        // Number of values per row in the `Colors` array.
+        public string ColorsStride { get; private set; }
+
+        //  Whether or not the individual vertices should be colored using the data in the corresponding arrays.
+        public bool RequiresPerVertexColoration { get; private set; }
+
+        private static string EncodeNumbers<T>(IEnumerable<T> coordinates)
+        {
+            var stream = new MemoryStream();
+            using (var writer = new BinaryWriter(stream))
+            {
+                if (typeof(T) == typeof(double))
+                {
+                    foreach (T value in coordinates)
+                        writer.Write(Convert.ToSingle(value));
```

Comment:
whats this about? CoGS format doesn't support doubles?

Reply:
The 'Cogs' format is storing doubles as floats as I read it. I think that is fine and not something that can be changed without breaking the format, right? This method is borrowed from reach.

Reply:
CoGS is fine with double precision.

Reply:
`float` in Swagger just means floating point, doesn't imply single precision.

Reply:
This is for the arrays that are encoded as ```base64``` strings. So they are ```strings``` in swagger. Is Cogs encoding these as ```double```? Or as ```float``` the same way reach is doing it? I have been assuming the latter.

Reply:
Ah, yes, those are 32 bit.